curated_transformers/util/string.py

```diff
@@ -0,0 +1,140 @@
+
+        :param string:
+            String to transform.
+        :returns:
+            Transformed string.
+        """
+        return self._apply(string)
+
+    def revert(self, string: str) -> str:
+        """
+        Reverts the previously applied transformation of the given string.
+
+        :param string:
+            Previously transformed string.
+        :returns:
+            Reverted string.
+        """
+        if self._reversible:
+            return self._revert(string)
+        else:
+            return string
+
+
+class StrSub(StringTransform):
+    """
+    Substitute a substring with another string.
+    """
+
+    def __init__(self, forward: Tuple[str, str], backward: Optional[Tuple[str, str]]):
+        """
+        Construct a reversible substitution.
+
+        :param forward:
+            Tuple where the first string is a RegEx pattern
+            and the second the replacement.
+
+            This operation is performed when the :meth:`.apply`
+            method is invoked.
+        :param backward:
+            Optional tuple where the first string is a RegEx pattern
+            and the second the replacement.
+
+            This operation is performed when the :meth:`.revert`
+            method is invoked. If ``None``, it is a no-op.
+        """
+        super().__init__(backward is not None)
+
+        self.forward = forward
+        self.backward = backward
+
+    def _apply(self, string: str) -> str:
+        return re.sub(self.forward[0], self.forward[1], string)
+
+    def _revert(self, string: str) -> str:
+        assert self.backward is not None
+        return re.sub(self.backward[0], self.backward[1], string)
+
+
+class StrSubInv(StrSub):
+    """
+    A substitution whose backward transformation can be
+    automatically derived from the forward transformation.
+    """
+
+    def __init__(self, forward: Tuple[str, str]):
+        """
+        Construct a reversible (and invertible) substitution.
+
+        :param forward:
+            Tuple where the first string is string to match
+            and the second the replacement, neither of which
+            can contain RegEx meta-characters.
+        """
+        super().__init__(
+            forward=(f"{re.escape(forward[0])}", forward[1]),
+            backward=(f"{re.escape(forward[1])}", forward[0]),
+        )
+
+
+class StrRepl(StrSub):
+    """
+    Replaces an entire string with another.
+    """
+
+    def __init__(self, replacee: str, replacement: str, *, reversible: bool = True):
+        """
+        Construct a reversible replacement.
+
+        :param replacee:
+            The full string to be replaced.
+        :param replacement:
+            The replacement string.
+        """
+        super().__init__(
+            forward=(f"^{re.escape(replacee)}$", replacement),
+            backward=(f"^{re.escape(replacement)}$", replacee) if reversible else None,
+        )
+
+
+class StrLStrip(StrSub):
```

Comment:
```suggestion
class StringRemovePrefix(StrSub):
```

lstrip is a bit confusing, because Python `lstrip` removes using a set of characters.

Reply:
Renamed.